src/code/z_actor.c

```diff
@@ -2771,26 +2795,36 @@ void Actor_FreeOverlay(ActorOverlay* actorOverlay) {
 
         if (actorOverlay->loadedRamAddr != NULL) {
             if (actorOverlay->allocType & ACTOROVL_ALLOC_PERSISTENT) {
+#if OOT_DEBUG
                 if (HREG(20) != 0) {
                     PRINTF("オーバーレイ解放しません\n"); // "Overlay will not be deallocated"
                 }
+#endif
             } else if (actorOverlay->allocType & ACTOROVL_ALLOC_ABSOLUTE) {
+#if OOT_DEBUG
                 if (HREG(20) != 0) {
                     // "Absolute magic field reserved, so deallocation will not occur"
                     PRINTF("絶対魔法領域確保なので解放しません\n");
                 }
```

Comment:
```suggestion
                if (OOT_DEBUG && (HREG(20) != 0)) {
                    // "Absolute magic field reserved, so deallocation will not occur"
                    PRINTF("絶対魔法領域確保なので解放しません\n");
                }
```
Worth noting I think this looks a lot cleaner, but doesn't match for most `HREG(20)`'s

Reply:
How about

```
#define ACTOR_PRINTF(...)  // or equivalent IDO hack \
    if (HREG(20) != 0) { \
        PRINTF(_VA_ARGS); \
    }
```

I checked that this works for all `HREG(20) != 0`s, even the ones you didn't ifdef out here

Reply:
Are you sure this works? I can't get `__VA_ARGS__` to work

Reply:
variadic macros do not exist in c89, and its why we had to do a weird hack for osSyncPrintf

Reply:
Maybe `#define ACTOR_PRINTF if (HREG(20) != 0) PRINTF`? Not pretty but it might be better than `ACTOR_PRINTF0`/`ACTOR_PRINTF1`/`ACTOR_PRINTF2`

Reply:
Yeah, it actually works. Just needed 1 extra `if (1) {}` to match, but you can see the last commit to see what it's like

Reply:
Nice!

Does the GCC check warn about these? We might want something like in macros.h
```
#else
#ifdef __sgi /* IDO compiler */
// See comment about PRINTF in macros.h
#define ACTOR_PRINTF(args) (void)0
#else
#define ACTOR_PRINTF(format, ...) (void)0
#endif
#endif
```

Reply:
or `#elif defined(__sgi)` to remove an `#endif`, I forgot about that when writing the PRINTF one